src/hooks/useModalClose.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+
+const useCloseModal = (showModal, handleClose, modalRef) => {
+  useEffect(() => {
+    const handleOutsideClick = (e) => {
+      if (modalRef.current && !modalRef.current.contains(e.target)) {
+        handleClose();
+      }
+    };
+    const handleEscapeKey = (e) => {
+      if (e.keyCode === 27) {
```

Comment:
```suggestion
      if (e.keyCode === 27) {
```
여기 주석으로 esc 키 코드라는 걸 남겨도 좋을 것 같네요!